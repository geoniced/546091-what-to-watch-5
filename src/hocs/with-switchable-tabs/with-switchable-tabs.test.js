import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withSwitchableTabs from "./with-switchable-tabs";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withSwitchableTabs(MockComponent);

it(`renders withSwitchableTabs HOC`, () => {
  const tree = renderer
    .create(
        <MockComponentWrapped>
          <React.Fragment />
        </MockComponentWrapped>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
