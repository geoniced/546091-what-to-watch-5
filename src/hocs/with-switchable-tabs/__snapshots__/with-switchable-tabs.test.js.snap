// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders withSwitchableTabs HOC 1`] = `<div />`;
